File: ios/Cast/CastStateResolver.swift
import Foundation

/// Pure mapping from the three observable Cast facts to the cross-platform
/// `CastState` (`connected` > `connecting` > `notConnected` > `noDevices`).
///
/// **Cast-SDK-free and UNGATED on purpose** — it touches only the generated
/// `CastState` enum (always present, independent of the Cast build flag), so it
/// compiles in the default build and is unit-testable from the SPM test target
/// (mirrors Android's `CastStateResolver`). The gated `CastStateMapper` reads the
/// three booleans off the GCK enums at the call boundary and delegates here.
enum CastStateResolver {
  static func resolve(connected: Bool, connecting: Bool, hasDevices: Bool) -> CastState {
    if connected { return .connected }
    if connecting { return .connecting }
    if hasDevices { return .notConnected }
    return .noDevices
  }
}

File: android/src/main/java/com/audiobrowser/cast/CastStateResolver.kt
package com.audiobrowser.cast

import com.margelo.nitro.audiobrowser.CastState

/**
 * Pure mapping from the three observable Cast-session facts to the cross-platform [CastState]
 * (CONNECTED > CONNECTING > NOT_CONNECTED > NO_DEVICES). Lives in **main** (touches only the
 * generated [CastState] enum, no Cast SDK) so it compiles in the default build and is unit-testable
 * from `src/test`. `CastSessionController.getState()` reads the booleans off the Cast SDK and
 * delegates here.
 */
object CastStateResolver {
  fun resolve(connected: Boolean, connecting: Boolean, hasDevices: Boolean): CastState =
    when {
      connected -> CastState.CONNECTED
      connecting -> CastState.CONNECTING
      hasDevices -> CastState.NOT_CONNECTED
      else -> CastState.NO_DEVICES
    }
}
